error: unexpected end of input, expected `from`
  --> ../tests/pass/omg.rs:29:1
   |
29 | #[extract_lemma]
   | ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `extract_lemma` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected `from`
  --> ../tests/pass/omg.rs:37:1
   |
37 | #[extract_lemma]
   | ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `extract_lemma` (in Nightly builds, run with -Z macro-backtrace for more info)

[TIMESTAMP WARN  lib_rtg::metadata] could not read metadata for crate `"std"`: Os { code: 2, kind: NotFound, message: "No such file or directory" }
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _5
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _9
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _7
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _3
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _2
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _1
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: ((*((*_1).0: *mut N<T>)).0: T)
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _3
[TIMESTAMP WARN  lib_rtg::codegen::block] Not handling drop properly yet: _2
error[E0425]: cannot find function `extract_x` in this scope
  --> ../tests/pass/omg.rs:90:13
   |
90 |             extract_x(self);
   |             ^^^^^^^^^------
   |             |
   |             help: try calling `extract_x` as a method: `self.extract_x()`

thread 'rustc' panicked at lib_rust_to_gil/src/codegen/body.rs:13:82:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: core::option::unwrap_failed
   4: core::option::Option<T>::unwrap
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/option.rs:932:21
   5: lib_rtg::codegen::body::<impl lib_rtg::codegen::context::GilCtxt>::push_alloc_local_decls::{{closure}}
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/codegen/body.rs:13:47
   6: core::iter::traits::iterator::Iterator::for_each::call::{{closure}}
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/traits/iterator.rs:815:29
   7: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/adapters/map.rs:89:21
   8: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold::enumerate::{{closure}}
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/adapters/enumerate.rs:108:27
   9: <core::slice::iter::Iter<T> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/slice/iter/macros.rs:230:27
  10: <core::iter::adapters::enumerate::Enumerate<I> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/adapters/enumerate.rs:114:9
  11: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/adapters/map.rs:129:9
  12: core::iter::traits::iterator::Iterator::for_each
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/core/src/iter/traits/iterator.rs:818:9
  13: lib_rtg::codegen::body::<impl lib_rtg::codegen::context::GilCtxt>::push_alloc_local_decls
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/codegen/body.rs:9:9
  14: lib_rtg::codegen::body::<impl lib_rtg::codegen::context::GilCtxt>::push_body
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/codegen/body.rs:99:9
  15: lib_rtg::prog_context::ProgCtx::compile_fn
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/prog_context.rs:67:24
  16: lib_rtg::prog_context::ProgCtx::final_prog
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/prog_context.rs:89:21
  17: lib_rtg::prog_context::ProgCtx::compile_prog
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/prog_context.rs:140:24
  18: <lib_rtg::callbacks::ToGil as rustc_driver_impl::Callbacks>::after_expansion::{{closure}}
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/callbacks.rs:76:17
  19: rustc_middle::ty::context::GlobalCtxt::enter::{{closure}}
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/compiler/rustc_middle/src/ty/context.rs:771:37
  20: rustc_middle::ty::context::tls::enter_context::{{closure}}
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  21: std::thread::local::LocalKey<T>::try_with
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/std/src/thread/local.rs:286:12
  22: std::thread::local::LocalKey<T>::with
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/library/std/src/thread/local.rs:262:9
  23: rustc_middle::ty::context::tls::enter_context
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  24: rustc_middle::ty::context::GlobalCtxt::enter
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/compiler/rustc_middle/src/ty/context.rs:771:9
  25: rustc_interface::queries::QueryResult<&rustc_middle::ty::context::GlobalCtxt>::enter
             at /rustc/8f9080db423ca0fb6bef0686ce9a93940cdf1f13/compiler/rustc_interface/src/queries.rs:70:9
  26: <lib_rtg::callbacks::ToGil as rustc_driver_impl::Callbacks>::after_expansion
             at /Users/sacha/Prog/work/rust-to-gil/lib_rust_to_gil/src/callbacks.rs:73:9
  27: rustc_interface::interface::run_compiler::<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/GillianPlatform/rust-to-gil

note: please make sure that you have updated to the latest nightly

note: please attach the file at `/Users/sacha/Prog/work/rust-to-gil/rust_to_gil/rustc-ice-2024-06-30T14_06_14-14772.txt` to your bug report

note: compiler flags: -Z crate-attr=feature(register_tool) -Z crate-attr=register_tool(gillian) -Z crate-attr=feature(rustc_attrs) -Z crate-attr=allow(internal_features) -Z crate-attr=feature(stmt_expr_attributes) --crate-type lib

query stack during panic:
end of query stack
error[E0425]: cannot find function `extract_y` in this scope
   --> ../tests/pass/omg.rs:100:13
    |
100 |             extract_y(self);
    |             ^^^^^^^^^------
    |             |
    |             help: try calling `extract_y` as a method: `self.extract_y()`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0425`.
